### PR TITLE
Update transport

### PR DESCRIPTION
**Pull request by pablogs9**
This PR modifies the transport waiting approach for serial usb

It also modifies QoS of sensors_apps in order to be compatible with RQt

### CODE REVIEW

File: microros_extensions/zephyr_usb_serial_transport.c [hunk 1]
```diff
@@ -130,22 +130,17 @@ size_t uxr_write_serial_data_platform(uxrSerialPlatform* platform, uint8_t* buf,
...
 
-	k_sleep(K_MSEC(timeout));
+	size_t read = 0;
+	int spent_time = 0;
 
-	uart_irq_rx_disable(platform->uart_dev);
+	while(ring_buf_is_empty(&in_ringbuf) && spent_time < timeout){
+		k_sleep(K_MSEC(1));
+		spent_time++;
+	}
```
Comment (julionce) [inline microros_extensions/zephyr_usb_serial_transport.c]: Please, fix the indentation.

File: apps/sensors_publisher/src/main.c [hunk 2]
```diff
@@ -65,23 +63,23 @@ void main(void)
 
 	// Creating TOF publisher
 	rcl_publisher_options_t tof_publisher_ops = rcl_publisher_get_default_options();
-	tof_publisher_ops.qos.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+	// tof_publisher_ops.qos.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
```
Comment (julionce) [inline apps/sensors_publisher/src/main.c]: Could it be removed?

File: apps/sensors_publisher/src/main.c [hunk 3]
```diff
@@ -65,23 +63,23 @@ void main(void)
...
 	rcl_publisher_t tof_publisher = rcl_get_zero_initialized_publisher();
 	RCCHECK(rcl_publisher_init(&tof_publisher, &node, ROSIDL_GET_MSG_TYPE_SUPPORT(std_msgs, msg, Float32), "/sensors/tof", &tof_publisher_ops))
 
 	// Creating IMU publisher
 	rcl_publisher_options_t imu_publisher_ops = rcl_publisher_get_default_options();
-	imu_publisher_ops.qos.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+	// imu_publisher_ops.qos.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
```
Comment (julionce) [inline apps/sensors_publisher/src/main.c]: Could it be removed?

**Review by julionce — CHANGES_REQUESTED** [review]

File: apps/sensors_publisher/src/main.c (hunk 3 above)
Comment (pablogs9) [reply apps/sensors_publisher/src/main.c]: Done

File: apps/sensors_publisher/src/main.c (hunk 2 above)
Comment (pablogs9) [reply apps/sensors_publisher/src/main.c]: Done

**Review by julionce — APPROVED** [review]

**Pull request merged**